vault/resource_ldap_auth_backend.go

```diff
@@ -171,6 +172,20 @@ func ldapAuthBackendUpdate(d *schema.ResourceData, meta interface{}) error {
 
 	path := ldapAuthBackendConfigPath(d.Id())
 	data := map[string]interface{}{}
+	if d.HasChange("tune") {
+		log.Printf("[INFO] Auth '%q' tune configuration changed", d.Id())
+		if raw, ok := d.GetOk("tune"); ok {
+			log.Printf("[DEBUG] Writing ldap auth tune to '%q'", d.Id())
+
+			err := authMountTune(client, "auth/"+d.Id(), raw)
+			if err != nil {
+				return nil
+			}
+
+			log.Printf("[INFO] Written ldap auth tune to '%q'", d.Id())
+			d.SetPartial("tune")
+		}
+	}
```

Comment:
I'm wondering if we should just detect changes rather than adding new attributes specifically for tuning. For example, in the nomad resource, we check if values have changed: https://github.com/hashicorp/terraform-provider-vault/blob/8637c22ebe4dcaa09b6bfe98295b38e954463fd0/vault/resource_nomad_secret_backend.go#L279-L292.

Reply:
Note that we only need to tune when we're changing parameters on the mount. For config changes, we can just send those to Vault via an update request.

Reply:
That's how it is done in resource_auth_backend.go. I tried to recreate that behaviour.

https://github.com/hashicorp/terraform-provider-vault/blob/c4d5ee7807f0a9b5516e6505a18cfe8bbddac8c6/vault/resource_auth_backend.go#L185-L199